polyply/src/backmap_dna.py

```diff
@@ -0,0 +1,268 @@
+    return frame
+
+
+def orient_template(strand, base, template, meta_frame):
+    """
+    Orient DNA bases
+
+    Parameters:
+    -----------
+    meta_molecule: :class:`polyply.src.meta_molecule`
+    current_node:
+        node key of the node in meta_molecule to which template referes to
+    template: dict[collections.abc.Hashable, np.ndarray]
+        dict of positions referring to the lower resolution atoms of node
+
+    Returns:
+    --------
+    dict
+        the oriented template
+    """
+
+    # Calculate intrinsic frame of the base
+    template_frame = _gen_base_frame(base, template)
+
+    # Detemine rotation matrices
+    inv_rot_template_frame = template_frame.T
+    rot_meta_frame = meta_frame
+
+    # Determine origin point of template
+    positions = np.array([template["N1"], template["N3"], template["C2"],
+                          template["C4"], template["C5"], template["C5"]])
+    ref_origin = center_of_geometry(positions)
+
+    # Write the template as array and center at origin
+    template_arr = np.zeros((3, len(template)))
+    key_to_ndx = {}
+    for ndx, key in enumerate(template.keys()):
+        template_arr[:, ndx] = template[key] - ref_origin
+        key_to_ndx[key] = ndx
+
+    # if base[:2] == "DG" or base[:2] == "DC":
+    #     rotation = Rotation.from_rotvec(np.pi * e2)
+    #     template_arr = rotation.apply(template_arr.T).T
+
+    # Transform base template into backward strand variant
+    if strand == "backward":
+        # Rotate 180 around e2 if backward strand
+        rotation = Rotation.from_rotvec(3 * np.pi * e2)
+        template_arr = rotation.apply(template_arr.T).T
+
+        # Rotate 180 around e3 if backward strand
+        rotation = Rotation.from_rotvec(np.pi * e3)
+        template_arr = rotation.apply(template_arr.T).T
+
+    # Rotate base frame into meta_molecule frame
+    template_rotated_arr = rot_meta_frame @ (inv_rot_template_frame @ template_arr)
+
+    # Translate rotated template along base-base vector
+    if strand == "backward":
+        template_final_arr = (template_rotated_arr.T + meta_frame[:, 1] * 0.3).T
+    else:
+        template_final_arr = (template_rotated_arr.T - meta_frame[:, 1] * 0.3).T
+
+    # Write the template back as dictionary
+    template_result = {}
+    for key, ndx in key_to_ndx.items():
+        template_result[key] = template_final_arr[:, ndx]
+
+    return template_result
+
+
+class Backmap_DNA(Processor):
+    """
+    This processor takes a a class:`polyply.src.MetaMolecule` and
+    places coordinates form a higher resolution createing positions
+    for the lower resolution molecule associated with the MetaMolecule.
+    """
+
+    def __init__(self, fudge_coords=0.4, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # self.fudge_coords = fudge_coords
+        self.fudge_coords = 1
+        self.rotation_angle = 0.59  # 34° in radian
+        self.closed = True
+
+    def _place_init_coords(self, meta_molecule):
```

Comment:
This function is too large for it's own good. For the proof of principle implementation this is fine but I'd recommend to put all things that have to do with the rotation frame into it's own linear algebra class and the use like.

```
frame = RotationFrame(reference_vector, is_closed, other_external_variables)
frame .rotate(position_array)
```